File: extension/src/background/tool-error.ts
/** 工具业务错误：WS tool_result 可带可选 code/details（协议 §3.3）。 */
export class ToolError extends Error {
  readonly code?: string;
  readonly details?: Record<string, unknown>;

  constructor(message: string, code?: string, details?: Record<string, unknown>) {
    super(message);
    this.name = 'ToolError';
    this.code = code;
    this.details = details;
  }
}

File: daemon/internal/ws/toolerror.go
package ws

// ToolError 扩展 tool_result 失败（协议 §3.3）：保留 error 字符串，可选 code/details。
type ToolError struct {
	Message string
	Code    string
	Details map[string]any
}

func (e *ToolError) Error() string { return e.Message }
